.github/workflows/conventional-commit.yml

```diff
@@ -0,0 +1,68 @@
+name: "Lint PR title"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+permissions:
+  pull-requests: write
+
+jobs:
+  main:
+    name: Validate PR titles
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 9.1.1
+      - name: Generate scopes
+        id: generate_scopes
+        run: |
+          scopes=$(pnpm m ls --json --depth=-1 | grep "name" | cut -d":" -f2 | grep "@novu" | cut -d"/" -f2 | grep "\"" | cut -d"\"" -f1)
+          echo 'SCOPES<<EOF' >> $GITHUB_ENV
+          echo "$scopes" >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+      - uses: amannn/action-semantic-pull-request@v5
+        id: lint_pr_title
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          scopes: |
+            ${{ env.SCOPES }}
+      - uses: marocchino/sticky-pull-request-comment@v2
+        # When the previous steps fails, the workflow would stop. By adding this
+        # condition you can continue the execution with the populated error message.
+        if: always() && (steps.lint_pr_title.outputs.error_message != null)
+        with:
+          header: pr-title-lint-error
+          message: |
+            Hey there and thank you for opening this pull request! 👋
+            
+            We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.
+
+            Your PR title is: `${{ github.event.pull_request.title }}`
+            It should be something like: `feat(scope): Add fancy new feature`
+
+            Details:
+            
+            ```
+            ${{ steps.lint_pr_title.outputs.error_message }}
+            ```
```

Comment:
A friendly message added as a PR comment when the PR title is not conventional, alongside the actions needed to fix it.

Reply:
This only affects the PR title? What about the squash commit message?

Reply:
It targets the PR title which in turn is used as the squash commit message (the highlighted part of this screenshot). The PR description is not targeted by this action.

This won't target unsquashed commit messages that may have been pushed. That job still lives with the Git-husky pre-commit linting hook. We'll need to enforce squash-merges on Git to ensure all commits on the tree are conventionally titled.